source/common/upstream/load_balancer_impl.h

```diff
@@ -410,6 +420,29 @@ class EdfLoadBalancerBase : public ZoneAwareLoadBalancerBase {
 
   // Scheduler for each valid HostsSource.
   absl::node_hash_map<HostsSource, Scheduler, HostsSourceHash> scheduler_;
+
+protected:
+  // Slow start related configs
+  const envoy::config::cluster::v3::Cluster::CommonLbConfig::EndpointWarmingPolicy
+      endpoint_warming_policy_;
+  const std::chrono::milliseconds slow_start_window_;
+  double time_bias_{1.0};
+  const std::unique_ptr<Runtime::Double> time_bias_runtime_;
+  TimeSource& time_source_;
+  struct orderByCreateDateDesc {
+    bool operator()(const HostSharedPtr l, const HostSharedPtr r) const {
+      return (l->creationTime() < r->creationTime()) ||
+             (l->address()->asString() < r->address()->asString());
```

Comment:
experimented with comparator a little bit and this one is optimal for ordering+unique host retrieval/removal (relies on host addresses (hostAndPort) being unique)